lisa/sut_orchestrator/azure/common.py

```diff
@@ -132,10 +148,81 @@ def marketplace(self, value: Optional[AzureVmMarketplaceSchema]) -> None:
+        shared_gallery: Optional[SharedImageGallerySchema] = self._shared_gallery
+        if not shared_gallery:
+            if isinstance(self.shared_gallery_raw, dict):
+                # Users decide the cases of image names,
+                #  the inconsistent cases cause the mismatched error in notifiers.
+                # The lower() normalizes the image names,
+                #  it has no impact on deployment.
+                self.shared_gallery_raw = dict(
+                    (k, v.lower()) for k, v in self.shared_gallery_raw.items()
+                )
+                shared_gallery = SharedImageGallerySchema.schema().load(  # type: ignore
+                    self.shared_gallery_raw
+                )
+                if not shared_gallery.subscription_id:
+                    shared_gallery.subscription_id = self.subscription_id
+                # this step makes shared_gallery_raw is validated, and
+                # filter out any unwanted content.
+                self.shared_gallery_raw = shared_gallery.to_dict()  # type: ignore
+            elif self.shared_gallery_raw:
+                assert isinstance(
+                    self.shared_gallery_raw, str
+                ), f"actual: {type(self.shared_gallery_raw)}"
+                # Users decide the cases of image names,
+                #  the inconsistent cases cause the mismatched error in notifiers.
+                # The lower() normalizes the image names,
+                #  it has no impact on deployment.
+                shared_gallery_strings = re.split(
+                    r"[/]+", self.shared_gallery_raw.strip().lower()
+                )
+                if len(shared_gallery_strings) == 4:
+                    shared_gallery = SharedImageGallerySchema(*shared_gallery_strings)
+                    # shared_gallery_raw is used
+                    self.shared_gallery_raw = shared_gallery.to_dict()  # type: ignore
+                elif len(shared_gallery_strings) == 3:
+                    shared_gallery = SharedImageGallerySchema()
+                    shared_gallery.subscription_id = self.subscription_id
+                    shared_gallery.image_gallery = shared_gallery_strings[0]
+                    shared_gallery.image_definition = shared_gallery_strings[1]
+                    shared_gallery.image_version = shared_gallery_strings[2]
+
+                    # shared_gallery_raw is used
+                    self.shared_gallery_raw = shared_gallery.to_dict()  # type: ignore
+                else:
+                    raise LisaException(
+                        f"Invalid value for the provided shared gallery "
+                        f"parameter: '{self.shared_gallery_raw}'."
+                        f"The shared gallery parameter should be in the format: "
+                        f"'<subscription_id>/<image_gallery>/<image_definition>/<image_version>' "
+                        f"or '<image_gallery>/<image_definition>/<image_version>'"
```

Comment:
How about if image_version is latest? How it's supported?

Reply:
<image_version> can be set to value `latest`